### PR TITLE
{lib}[GCCcore/11.2.0] fmt v8.1.1

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+source_urls = ['https://github.com/fmtlib/fmt/releases/download/%(version)s/']
+sources = ['fmt-%(version)s.zip']
+checksums = ['23778bad8edba12d76e4075da06db591f3b0e3c6c04928ced4a7282ca3400e5d']
+
+builddependencies = [
+    ('CMake', '3.22.1'),
+    ('binutils', '2.37')
+]
+
+separate_build_dir = True
```
Comment (jfgrimm) [inline easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:21]: this is already the default in `CMakeMake`
```suggestion
separate_build_dir = True
```

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
```
Comment (jfgrimm) [inline easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:22]:
```suggestion

# build static and shared libraries
configopts = ["-DCMAKE_POSTITION_INDEPENDENT_CODE=TRUE", "-DBUILD_SHARED_LIBS=TRUE"]

runtest = 'test'

```

Comment (jfgrimm) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:22]: might as well build the shared library as well, and run the tests

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+easyblock = 'CMakeMake'
+
+name = 'fmt'
+version = '8.1.1'
+
+homepage = 'http://fmtlib.net/'
+description = "fmt (formerly cppformat) is an open-source formatting library."
+
+toolchain = {'name': 'GCCcore', 'version': '11.2.0'}
+toolchainopts = {'pic': True}
```
Comment (jfgrimm) [inline easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:10]: not needed when configuring with `-DCMAKE_POSTITION_INDEPENDENT_CODE=TRUE`
```suggestion
toolchainopts = {'pic': True}
```

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+builddependencies = [
+    ('CMake', '3.22.1'),
+    ('binutils', '2.37')
+]
+
+separate_build_dir = True
+
+sanity_check_paths = {
+    'files': ['lib/libfmt.a'],
```
Comment (jfgrimm) [inline easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:24]:
```suggestion
    'files': ['lib/libfmt.%s' % x for x in ['a', SHLIB_EXT]],
```

**Review by jfgrimm — DISMISSED** [review]
made some suggestions to enable both shared and static libraries to be built

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb (hunk 1 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:21]: That is correct. I only updated the previous EC file and overlooked that line. Thanks.

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb (hunk 2 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:22]: I agree. However,  thought this is what:
`toolchainopts = {'pic': True}`
does. Is that not the case?

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb (hunk 3 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:10]: As above. What is the 'better' way of doing it? I am sure something has slipped my attention here.

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb (hunk 4 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:24]: Changed. Thanks!

File: easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb (hunk 2 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:22]: good point, you could also leave the toolchainopts and use `configopts = ["", "-DBUILD_SHARED_LIBS=TRUE"]`

Comment (sassy-crick) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:22]: Is there a preferred way of doing this or is that just a matter of taste?

Comment (jfgrimm) [reply easybuild/easyconfigs/f/fmt/fmt-8.1.1-GCCcore-11.2.0.eb:22]: either should be fine

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3100.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/5e0ec183ba1fb1d20e89bb1cdc15ebe9 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**